src/udev/udevd.c

```diff
@@ -1490,9 +1496,26 @@ static int on_post(sd_event_source *s, void *userdata) {
         if (manager->exit)
                 return sd_event_exit(manager->event, 0);
 
-        if (manager->cgroup)
-                /* cleanup possible left-over processes in our cgroup */
-                (void) cg_kill(SYSTEMD_CGROUP_CONTROLLER, manager->cgroup, SIGKILL, CGROUP_IGNORE_SELF, NULL, NULL, NULL);
+        if (manager->cgroup && udev_ctrl_is_connected(manager->ctrl) <= 0) {
```

Comment:
doesn't this mean if a user is calling udevadm, it will also skip the cleanup? is it still necessary after the pid matching logic was introduced?